### PR TITLE
feat: overriding system value for PostgreSQL's identity column

### PR DESCRIPTION
**Pull request by amakmurr**
Add `OVERRIDING SYSTEM VALUE` insert clause when using PostgreSQL with version >= 10 dialect. It's necessary for table with always generated identity column also as primary key. By default those column must be always generated by PostgreSQL, but in this fixtures mostly case we want to define ourself `id` especially when declaring the relationship between fixtures.

This PR will allow you to force define value for column that has been made with such as statement `id BIGSERIAL GENERATED ALWAYS AS IDENTITY PRIMARY KEY`

Reference:
- https://www.postgresql.org/docs/current/sql-insert.html#:~:text=for%20UPDATE).-,OVERRIDING%20SYSTEM%20VALUE,-If%20this%20clause

### CODE REVIEW

File: postgresql.go [hunk 1]
```diff
@@ -383,3 +397,79 @@ func (*postgreSQL) quoteKeyword(s string) string {
...
+		return hasIdentityColumn, nil
+	}
+
+	parts := strings.Split(tableName, ".")
+	tableName = parts[0]
+	if len(parts) > 1 {
+		tableName = parts[1]
+	}
+	query := fmt.Sprintf(`
+		SELECT count(*) AS count
```
Comment (andreynering) [inline postgresql.go]:
```suggestion
		SELECT COUNT(*) AS count
```

File: postgresql.go [hunk 2]
```diff
@@ -383,3 +397,79 @@ func (*postgreSQL) quoteKeyword(s string) string {
...
+		return false, err
+	}
+
+	h.tablesHasIdentityColumn[tableName] = count > 0
+	return h.tablesHasIdentityColumn[tableName], nil
+}
+
+func (h *postgreSQL) getMajorVersion(q queryable) (int, error) {
+	var version string
+	err := q.QueryRow("SELECT version()").Scan(&version)
```
Comment (andreynering) [inline postgresql.go]:
```suggestion
	err := q.QueryRow("SELECT VERSION()").Scan(&version)
```

File: postgresql.go [hunk 3]
```diff
@@ -383,3 +397,79 @@ func (*postgreSQL) quoteKeyword(s string) string {
...
+	versionNumbers := re.FindAllString(version, -1)
+	if len(versionNumbers) > 0 {
+		majorVersion, err := strconv.Atoi(versionNumbers[0])
+		if err != nil {
+			return 0, err
+		}
+		return majorVersion, nil
+	}
+
+	return 0, fmt.Errorf("could not parse major version from: %s", version)
```
Comment (andreynering) [inline postgresql.go]:
```suggestion
	return 0, fmt.Errorf("testfixtures: could not parse major version from: %s", version)
```

**Review by andreynering — COMMENTED** [review]
Hi @amakmurr,

The CI failed because of an out-of-date dependency. I just updated it in master.

Locally, I noticed that tests for MySQL broke with you change. Can you rebase with master and fix the issue?

**Comment by amakmurr** [comment]
Hi @andreynering ,
I am just updating this PR and apply your suggestion. Also has been run the unit test for mysql in my local machine and it's works now. Please kindly take another look. 

Thank you.

**Review by andreynering — APPROVED** [review]

**Comment by andreynering** [comment]
Thanks @amakmurr!

I did some very simple improvements at b6a9478065da83965563b1196a1889b6b6dabf61.

**Pull request merged**